src/v/coproc/tests/partition_tests.cc

```diff
@@ -0,0 +1,157 @@
+#include "coproc/partition.h"
+#include "coproc/partition_manager.h"
+#include "coproc/tests/fixtures/redpanda_with_copro_fixture.h"
+#include "model/metadata.h"
+#include "storage/tests/utils/random_batch.h"
+#include "storage/types.h"
+#include "test_utils/fixture.h"
+
+#include <seastar/core/coroutine.hh>
+
+using namespace std::chrono_literals; // NOLINT
+
+namespace {
+class sleep_once_consumer {
+public:
+    explicit sleep_once_consumer(model::timeout_clock::duration d) noexcept
+      : _timeout(d) {}
+
+    ss::future<ss::stop_iteration> operator()(const model::record_batch& rb) {
+        if (_first_run) {
+            _first_run = false;
+            _started.set_value();
+            co_await ss::sleep(_timeout);
+        }
+        co_return ss::stop_iteration::no;
+    }
+
+    void end_of_stream() {}
+
+    /// Used to know when actual consumption will begin
+    ss::future<> started() { return _started.get_future(); }
+
+private:
+    bool _first_run{true};
+    ss::promise<> _started;
+    model::timeout_clock::duration _timeout;
+};
+
+model::topic_namespace to_tpns(const model::ntp& ntp) {
+    return model::topic_namespace(model::topic_namespace_view{ntp});
+}
+
+template<typename T>
+ss::future<bool>
+partition_exists(ss::sharded<T>& pm, ss::shard_id shard, model::ntp ntp) {
+    return pm.invoke_on(
+      shard, [ntp](T& local_pm) { return (bool)local_pm.get(ntp); });
+}
+
+ss::future<model::timeout_clock::duration> timed_delete_partition(
+  model::timeout_clock::duration timeout,
+  cluster::topics_frontend& topics_frontend,
+  coproc::partition_manager& local,
+  model::ntp tp2_ntp) {
+    auto partition = local.get(tp2_ntp);
+    BOOST_REQUIRE(partition);
+    /// Begin by writing some sample data into the partition/log
+    auto batches = storage::test::make_random_batches(model::offset{0});
+    const storage::log_append_config write_cfg{
+      .should_fsync = storage::log_append_config::fsync::no,
+      .io_priority = ss::default_priority_class(),
+      .timeout = model::no_timeout};
+    auto rbr = model::make_memory_record_batch_reader(std::move(batches));
+    auto rs = co_await std::move(rbr).for_each_ref(
+      partition->make_appender(write_cfg), model::no_timeout);
+    auto reader_cfg = storage::log_reader_config(
+      model::offset{0},
+      model::model_limits<model::offset>::max(),
+      ss::default_priority_class());
+    rbr = co_await partition->make_reader(reader_cfg);
+
+    /// Start the tests by consuming the pushed data, using the
+    /// sleep_once_consumer to hold up consumption for at least some known
+    /// duration
+    auto soc = sleep_once_consumer(timeout);
+    auto start_consume = soc.started();
+    auto finished = std::move(rbr).consume(std::move(soc), model::no_timeout);
+    /// To avoid deletion occuring before the stream actually started, wait on
+    /// the 'start_consume' future to known when consume has started
+    co_await std::move(start_consume);
```

Comment:
I'm wondering if these unit tests are useful anymore. They were supposed to shutdown the `gated_shutdown_reader` but they work without it, and now that this has been removed it seems like this just tests that the `coproc::partition_manager` waits on a `coproc::partition::gate` before closing. Maybe a bit overkill to test a simple thing?